### PR TITLE
Removed PMA_TEST_HEADERS checking as they have been defined

### PR DESCRIPTION
**Pull request by osaid-r**
Before submitting pull request, please check that every commit:

- [X] Has proper Signed-Off-By
- [X] Has commit message which describes it
- [X] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [X] Any new functionality is covered by tests

Signed-off-by : Osaid <osaid.nasir@gmail.com>

### CODE REVIEW

**Comment by ragnerok** [comment]
@nijel  why is this causing an error ? All I've done is remove the if conditions which were useless as the header tests were defined and hence not needed

**Comment by nijel** [comment]
You did not mock the Response::getInstance, so the function is working on real Response object instead of the mocked one...

**Comment by ragnerok** [comment]
So basically I have to mock response object in both functions for this to work right like done [here](https://github.com/phpmyadmin/phpmyadmin/commit/2148e7cad879a9edbc6a82ab03115cd0957f9bee#diff-7cd1872ccc0c19cfaf814d634989604aR370)

**Comment by nijel** [comment]
Yes

**Comment by ragnerok** [comment]
I still do not get why it wasn't showing an error for this earlier when the if conditions were still there but anyways thanks

**Comment by nijel** [comment]
Prior to the change the PHP's `header()` function has been mocked by runkit (or this was not tested on PHP 7 where runkit is not available), so it didn't actually sent headers...

**Comment by ragnerok** [comment]
Also I think I have written same response mock code in 2-3 files and it would have been better if I had made some kind of function for it

**Comment by ragnerok** [comment]
ok then I'll add a function responseMock with 2 arguments 1 is an array of strings for arguments for header() and $n which is the number of calls of header function to ConfigFile.php

**Comment by nijel** [comment]
I think it would better fit to test/PMATestCase.php...

**Comment by ragnerok** [comment]
Yeah makes sense
Thanks

File: test/libraries/core/PMA_headerLocation_test.php [hunk 1]
```diff
@@ -65,16 +66,14 @@ public function testSendHeaderLocationWithoutSidWithIis()
 
         $testUri = 'https://example.com/test.php';
 
-        $header = array('Location: ' . $testUri);
         PMA_sendHeaderLocation($testUri); // sets $GLOBALS['header']
-        $this->assertEquals($header, $GLOBALS['header']);
+        PMATestCase::mockResponse('Location: ' . $testUri);
```
Comment (nijel) [inline test/libraries/core/PMA_headerLocation_test.php]: I think it's better to base the test class on PMATestCase and use it as $this->mockResponse....

Comment (ragnerok) [inline test/libraries/core/PMA_headerLocation_test.php]: you mean PMA_HeaderLocation_Test should extend PMATestCase ?
but it already inherits from phpunit and multiple inheritance isn't allowed in php I think

Comment (nijel) [inline test/libraries/core/PMA_headerLocation_test.php]: But PMATestCase does extend PHPUnit_Framework_TestCase, so there is no problem with that :-)

Comment (ragnerok) [inline test/libraries/core/PMA_headerLocation_test.php]: oh I forgot about that nice !

File: test/PMATestCase.php [hunk 2]
```diff
@@ -21,4 +21,33 @@ public static function setUpBeforeClass()
...
+    /**
+     * Creates mock of Response object
+     *
+     * @param string $param parameter for header method
+     *
+     * @return void
+     */
+    public static function mockResponse($param)
+    {
+        $restoreInstance = PMA\libraries\Response::getInstance();
```
Comment (nijel) [inline test/PMATestCase.php]: You do not restore the instance anywhere...

File: test/libraries/core/PMA_headerLocation_test.php [hunk 3]
```diff
@@ -61,29 +61,18 @@ public function setUp()
...
 
-            //reset $GLOBALS['header'] for the next assertion
-            unset($GLOBALS['header']);
+        $testUri = 'https://example.com/test.php';
 
-            $header = array('Refresh: 0; ' . $testUri);
-            PMA_sendHeaderLocation($testUri, true); // sets $GLOBALS['header']
-            $this->assertEquals($header, $GLOBALS['header']);
+        PMA_sendHeaderLocation($testUri); // sets $GLOBALS['header']
+        $this->mockResponse('Location: ' . $testUri);
```
Comment (nijel) [inline test/libraries/core/PMA_headerLocation_test.php]: The mock should be before function call....

File: test/libraries/core/PMA_headerLocation_test.php [hunk 4]
```diff
@@ -171,4 +150,4 @@ public function testSendHeaderLocationIisLongUri()
 
         $attrInstance->setValue($restoreInstance);
```
Comment (nijel) [inline test/libraries/core/PMA_headerLocation_test.php]: The value is not set here.

File: test/libraries/core/PMA_headerLocation_test.php [hunk 5]
```diff
@@ -61,29 +61,18 @@ public function setUp()
...
+        PMA_sendHeaderLocation($testUri); // sets $GLOBALS['header']
+        $this->mockResponse('Location: ' . $testUri);
 
-        } else {
-            $this->markTestSkipped(
-                'Cannot redefine constant/function - missing runkit extension'
-            );
-        }
+        //reset $GLOBALS['header'] for the next assertion
+        unset($GLOBALS['header']);
```
Comment (nijel) [inline test/libraries/core/PMA_headerLocation_test.php]: No longer needed...

**Review by nijel — CHANGES_REQUESTED** [review]

File: test/libraries/core/PMA_headerLocation_test.php (hunk 3 above)
Comment (ragnerok) [inline test/libraries/core/PMA_headerLocation_test.php]: sorry I just realised my mistake

File: test/libraries/core/PMA_headerLocation_test.php (hunk 4 above)
Comment (ragnerok) [inline test/libraries/core/PMA_headerLocation_test.php]: this wasn't done by me but I'll correct it nonetheless

Comment (nijel) [inline test/libraries/core/PMA_headerLocation_test.php]: Ah, sorry, it works here because you're not using mockResponse in this method...

Comment (ragnerok) [inline test/libraries/core/PMA_headerLocation_test.php]: ok, but I was wondering is there a way to make use of mockResponse here as there is only the disable method that needs to be added

**Comment by nijel** [comment]
Now the problem is that you're not restoring the original Response object after tests. Probably the mockResponse should return it and there should be method to restore it at the end of test...

**Comment by ragnerok** [comment]
So the function should return $mockRespnse and then attrInstance 's value should be set to $mockResponse

**Comment by nijel** [comment]
You need to call `$attrInstance->setValue($restoreInstance);` in the end. Probably it should be again wrapped by some method. Maybe the mockResponse should store both parameters as attributes. Or even better, if these attributes are set, the mock response would be automatically restored in tearDown method.

**Comment by ragnerok** [comment]
why not make `$attrInstance` and `$mockResponse` as properties of PMATstCase then both are accessible using `$this`

**Comment by nijel** [comment]
Yes, that is what I was thinking about:

* add these properties to the class
* set them in mockResponse
* add `tearDown` method (which is executed after each test) which will restore original instance and clear the attributes if they were set

This way it will work properly even in case of test failure.

File: test/libraries/core/PMA_headerLocation_test.php [hunk 6]
```diff
@@ -171,4 +149,18 @@ public function testSendHeaderLocationIisLongUri()
 
         $attrInstance->setValue($restoreInstance);
     }
-}
+    /**
+     *Tear down function for mockResponse method
+     *
+     *@return void
+     */
+    public function tearDown()
```
Comment (nijel) [inline test/libraries/core/PMA_headerLocation_test.php]: Shouldn't the method rather go to the PMATestCase class?

Comment (ragnerok) [inline test/libraries/core/PMA_headerLocation_test.php]: Then the method will have to be protected but yeah that will be better

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**